lib/spack/spack/fetch_strategy.py

```diff
@@ -1244,10 +1237,13 @@ def _from_merged_attrs(fetcher, pkg, version):
     """Create a fetcher from merged package and version attributes."""
     if fetcher.url_attr == 'url':
         url = pkg.url_for_version(version)
+        mirrors = [spack.url.substitute_version(u, version)
+                   for u in getattr(pkg, 'urls', [])]
+        attrs = {fetcher.url_attr: url, 'mirrors': mirrors}
```

Comment:
I think it's worth adding `urls_for_version` or something like that, so that someone doesn't duplicate this logic later.

Reply:
I added a `TODO` but will extract a function if you feel strongly about it. I'd like to start working on a refactor immediately after this PR so I can do the work there (since probably extracting a function right now won't be very likely final).